lib/scholar/metrics.ex

```diff
@@ -1205,6 +1226,100 @@ defmodule Scholar.Metrics do
     1 - Nx.sum(weights_matrix * cm) / Nx.sum(weights_matrix * expected)
   end
 
+  @doc """
+
+  Top-k Accuracy classification score.
+
+  This metric computes the number of times where the correct label is among the top k labels predicted (ranked by predicted scores).
+
+  For binary task assumed that y_score have values from 0 to 1.
+
+  ## Options
+
+  #{NimbleOptions.docs(@top_k_accuracy_score_schema)}
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([0, 1, 2, 2, 0])
+      iex> y_score = Nx.tensor([[0.5, 0.2, 0.1], [0.3, 0.4, 0.5], [0.4, 0.3, 0.2], [0.1, 0.3, 0.6], [0.9, 0.1, 0.0]])
+      iex> Scholar.Metrics.top_k_accuracy_score(y_true, y_score, k: 2, num_classes: 3)
+      #Nx.Tensor<
+        f32
+        0.800000011920929
+      >
+
+      iex> y_true = Nx.tensor([0, 1, 2, 2, 0])
+      iex> y_score = Nx.tensor([[0.5, 0.2, 0.1], [0.3, 0.4, 0.5], [0.4, 0.3, 0.2], [0.1, 0.3, 0.6], [0.9, 0.1, 0.0]])
+      iex> Scholar.Metrics.top_k_accuracy_score(y_true, y_score, k: 2, num_classes: 3, normalize: false)
+      #Nx.Tensor<
+        u64
+        4
+      >
+
+      iex> y_true = Nx.tensor([0, 1, 0, 1, 0])
+      iex> y_score = Nx.tensor([0.55, 0.3, 0.1, -0.2, 0.99])
+      iex> Scholar.Metrics.top_k_accuracy_score(y_true, y_score, k: 1, num_classes: 2)
+      #Nx.Tensor<
+        f32
+        0.20000000298023224
+      >
+  """
+
+  deftransform top_k_accuracy_score(y_true, y_prob, opts \\ []) do
+    top_k_accuracy_score_n(
+      y_true,
+      y_prob,
+      NimbleOptions.validate!(opts, @top_k_accuracy_score_schema)
+    )
+  end
+
+  defnp top_k_accuracy_score_n(y_true, y_score, opts) do
+    k = opts[:k]
+    num_classes = opts[:num_classes]
+
+    hits =
+      case num_classes do
+        1 ->
+          raise ArgumentError, "num_classes must be greater than 1"
+
+        2 ->
+          if k == 1 do
+            rank_1d(Nx.rank(y_score))
+            threshold = 0.5
+            y_pred = y_score > threshold
+            y_pred == y_true
+          else
+            Nx.broadcast(Nx.u8(1), y_true)
+          end
+
+        _ ->
+          check_num_classes(num_classes, Nx.axis_size(y_score, 1))
+          sorted_pred = Nx.argsort(y_score, axis: 1, direction: :desc)
+
+          Nx.any(Nx.new_axis(y_true, 0) == Nx.transpose(sorted_pred[[.., 0..(k - 1)//1]]),
+            axes: [0]
+          )
+      end
+
+    case opts[:normalize] do
+      true -> Nx.mean(hits)
+      false -> Nx.sum(hits)
+    end
```

Comment:
Can this function be used anywhere? https://hexdocs.pm/nx/Nx.html#top_k/2

Reply:
If this function allows us to choose the axis then yes. But currently, I need to use `Nx.argsort` with the axis set to 1.

https://github.com/msluszniak/scholar/blob/9854fbc25a9fb9cddf0947aa4113c51639cf74f9/lib/scholar/metrics.ex#L1295C1-L1300